### ISSUE TITLE
PlantUML integration

### ISSUE THREAD

**Issue by silverwind** [go-gitea/gitea#18272]
### Feature Description

There are customization docs that instruct the user to insert some JS into their templates for PlantUML support. I think we can do better and add configuration options to set a URL to a PlantUML server instead and have our own JS handle the plumbing to get the graphs rendered.

Related: https://docs.gitlab.com/ee/administration/integration/plantuml.html
Also related (but not required): https://github.com/alecthomas/chroma/issues/580

Ideally, the rendering would be done completely in JS without a dependency on a server but I don't think such a module exists (yet).

**Comment by lunny** [comment]
This may become a plugin after #34917 is implemented.

**Comment by silverwind** [comment]
As per https://github.com/go-gitea/gitea/issues/36747, it would be nice to also be able to render inline PlantUML syntax in Markdown, e.g. just how it's already done by Mermaid.

I think this requires deep integration in the UI and probably can or should not be done with a plugin, so it could be opt-in by the user with a config option that declares their external server URL.
